### PR TITLE
Thread Previews with Truncated Images and Modal Viewer

### PR DESCRIPTION
**Pull request by salman-neslit**
## Link to Issue
Closes: #9455

## Description of Changes
- Implemented truncated image preview in the thread preview if a thread contains an image.
- Enabled clicking on the thread text to navigate directly to the thread (maintained existing behavior).
- Made the image clickable to open in a dedicated view, with the thread content.

## "How We Fixed It"
- For the truncated image preview, we added CSS class to limit the image display size within the thread preview.
- Adjusted the onClick event for the image to open a separate view, showing the full image and thread content side-by-side.

## Test Plan
- Verified truncated image preview on thread preview.
- Confirmed redirection to thread upon text click.
- Confirmed image opens in a new view with side-by-side thread content.

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  .modal-content {
+    position: relative;
+    display: flex;
+    width: 80%;
+    height: 80%;
+    max-width: 1200px;
+    max-height: 90vh;
+    background-color: #fff;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss:29]: use color variables

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    border-radius: 8px;
+    overflow: hidden;
+
+    .modal-left {
+      position: relative;
+      width: 50%;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      background-color: #f4f4f4;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss:39]: same

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      .modal-image {
+        width: 100%;
+        height: auto;
+        object-fit: cover;
+      }
+
+      .nav-btn {
+        position: absolute;
+        background: rgba(0, 0, 0, 0.5);
+        color: #fff;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.scss:68]: and everywhere  else

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+import Permissions from '../../../../scripts/utils/Permissions';
+import { CWIcon } from '../../components/component_kit/cw_icons/cw_icon';
+import { ThreadCard } from '../../pages/discussions/ThreadCard';
+import './ThreadPreviewModal.scss';
+
+type ThreadModalProps = {
+  thread: Thread;
+};
+
+const ThreadModal: React.FC<ThreadModalProps> = ({ thread }) => {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx:21]: nit: we dont use FC notation

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  };
+
+  const handlePrevious = () => {
+    setCurrentImageIndex(
+      (prevIndex) => (prevIndex - 1 + images.length) % images.length,
+    );
+  };
+
+  return (
+    <div className="thread-modal">
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx:139]: ThreadPreviewModal

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+      hideReactionButton
+      hideUpvotesDrawer
+      layoutType="author-first"
+      showCommentState
+      removeImagesFromMarkDown
+    />
+  );
+};
+
+type FeedThreadPreviewProps = {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx:110]: ThreadPreviewModalProps

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  thread: Thread;
+  isThreadModalOpen: boolean;
+  onClose: () => void;
+  images: string[];
+};
+
+// eslint-disable-next-line react/no-multi-comp
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx:117]: we added this rule to obey it, not to comment it out. Please create separate file for each component

File: packages/commonwealth/client/scripts/views/components/feed.tsx [hunk 8]
```diff
@@ -23,11 +24,18 @@ import {
...
 
 const DEFAULT_COUNT = 10;
 
-const FeedThread = ({ thread }: { thread: Thread }) => {
+type FeedThreadProps = {
+  thread: Thread;
+  onClick: () => void;
+};
+
+const FeedThread: React.FC<FeedThreadProps> = ({ thread, onClick }) => {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/feed.tsx:38]: as I mentioned previously, we dont use FC in the project so please keep the consistency

File: packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  const handlePrevious = () => {
+    setCurrentImageIndex(
+      (prevIndex) => (prevIndex - 1 + images.length) % images.length,
+    );
+  };
+
+  return (
+    <div className="thread-preview-modal">
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/ThreadPreviewModal/ThreadPreviewModal.tsx:36]:
```suggestion
    <div className="ThreadPreviewModal">
```

**Review by masvelio — APPROVED** [review]
I doesnt break at least, but I guess it is not ideal mobile solution. It would require design team input. 

https://github.com/user-attachments/assets/8b0d1786-9a13-4cd8-a07b-dc6ab8f307a4

For now I am ok with it

**Review by mzparacha — APPROVED** [review]
Works well, also seems like we only want to show the image preview on the global feed, and not in the /overview and /discussion pages of a thread (cc: @salman-neslit @Israellund to confirm). 

The one on the global feed lgtm.

**Comment by Israellund** [comment]
@sachben91 @sssssabinaaa Can you weigh in on this? "Works well, also seems like we only want to show the image preview on the global feed, and not in the /overview and /discussion pages of a thread"
@masvelio I will make a ticket to clean up the mobile with new designs.

**Comment by Israellund** [comment]
@salman-neslit new mobile designs 
https://www.figma.com/design/bP5OITAQuDGgC9wkvXwX6h/Community-Wallet?node-id=396-7838&t=iwKL20Grvso93rWi-1

**Comment by salman-neslit** [comment]
Could you kindly review the recent changes, specifically the commit `2097a1a3df70db1e36a399172ab9a5c36b5a9cdf`? I would appreciate your feedback on whether the implementation is aligned with best practices, or if there are any suggestions for improvement.

**Pull request merged**